src/overlays/actors/ovl_En_Aob_01/z_en_aob_01.c

```diff
@@ -885,40 +1050,46 @@ void func_809C2F34(EnAob01* this, PlayState* play) {
     this->actor.prevPos = this->actor.world.pos;
 }
 
-void func_809C2FA0(void) {
+/**
+ * Generates a randomized list of dog text offsets and stores them in weekEventRegs.
+ * EnDg uses these text offsets to determine what text to show when the player picks
+ * the dog up while equipped with the Mask of Truth, and EnRacedog uses them to
+ * determine what condition the dog is in.
+ */
+void EnAob01_InitializeDogTextOffsets(void) {
     u8 i;
     u8 rand;
-    u8 index;
-    u8 orig;
-    u8 orig2;
-    u8 sp44[7];
-    u8 sp34[] = {
+    u8 dogTextOffsetsTemp;
+    u8 packedDogTextOffsets[7];
+    u8 dogTextOffsets[] = {
         0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13,
     };
```

Comment:
```suggestion
    u8 packedDogTextOffsets[RACEDOG_COUNT / 2];
    u8 dogTextOffsets[RACEDOG_COUNT] = {
        0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13,
    };
```
Then you could do something like this.

Reply:
This is clever, but unfortunately doesn't function properly in practice. The code makes the assumption that the number of dogs is even. It is particularly this code that is problematic:
```c
        if (i % 2) {
            packedDogTextOffsets[j] |= dogTextOffsetsTemp << 4;
            SET_WEEKEVENTREG_DOG_RACE_TEXT_OFFSETS(j, packedDogTextOffsets[j]);
        } else {
            packedDogTextOffsets[j] |= dogTextOffsetsTemp;
        }
```
Imagine, for example, that you bumped the number of dogs from 14 to 15. This means the indices will be 0, 1, 2, 3, ..., 12, 13, and 14. What happens when the index is 14? It will go to the second part of that if-else block and set the value in `packedDogTextOffsets`, but since there is no index 15 to follow, it will *never* actually write this to the weekEventRegs. I don't think this is fixable without breaking matching.

Given this, do you still want to use `RACEDOG_COUNT` here? Maybe with a comment that this function as-is only works when it's even?

Reply:
Oh that's so silly, so yeah it has to be even. I think I'd still use `RACEDOG_COUNT` with a note like this if I understand correctly? (can also write a better comment too)
```c
        if (i % 2) {
            packedDogTextOffsets[j] |= dogTextOffsetsTemp << 4;
            //! @note sets text offsets for both even and odd entry.
            //! Requires `RACEDOG_COUNT` to be even to work properly
            SET_WEEKEVENTREG_DOG_RACE_TEXT_OFFSETS(j, packedDogTextOffsets[j]);
        } else {
            packedDogTextOffsets[j] |= dogTextOffsetsTemp;
        }
```